main-application/comms-processing_tasks.c

```diff
@@ -0,0 +1,72 @@
+/*  comms-processing_tasks.c
+
+    Created by Minh Phong Bill Truong on April 28th 2020
+ */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "semphr.h"
+
+#include <stdio.h>
+#include "comms-processing_tasks.h"
+
+//==============================================================================
+//                                   GLOBALS
+//==============================================================================
+/**
+ * This Queue is used to store received and unprocessed packets
+ */
+QueueHandle_t rawPacketReceiveQueue;
+
+/**
+ * This Queue is used to store deparsed packets that will be used by transmitTask
+ */
+QueueHandle_t packetsToTransmit;
+
+//==============================================================================
+//                                  FUNCTIONS
+//==============================================================================
+
+/**
+ *  @brief Call functions required to deparse packet
+ *  
+ *  The deparser task receives the data that is to be sent down (clarify) and calls the
+ *  required functions to handle the formatting of the data into a packet. 
+ * 
+ *  @header	"software-and-command/main-applicatoion/comms-processing_tasks.c"
+ *	@param	Unused
+ *	@pre	None
+ * 	@post	Formatted packet is stored in memory
+ * 	@return	None
+ */
+void deparseTask(void* pvparameters) 
```

Comment:
How do you feel about the name deparse? Personally I hate it even though I think I was the one who coined it haha. How about encodeTask?